### ISSUE TITLE
Updating the generator using the joint loss gradient?

### ISSUE THREAD

**Issue by 1420108** [tadax/glcic#2]
Hi, thank you very much for your work. In the paper, they use the joint loss (L2 and GAN loss) gradient to update the generator G, but in your implementation, I notice that you only use L2 loss gradient to update G. Have I missed something here? Thank you.

**Comment by tadax** [comment]
Thank you for your comment! You are right. I will update and re-train the model in a few days.

**Comment by 1420108** [comment]
@tadax , thanks for your reply. As there's no direct connection between the Completion network and the Discriminators, I don't know how to update the Completion Network using the Joint Loss Gradients (Completion Network output: 256x256x3, Local and Global Discriminator input: 128x128x3). How can you backward the gradients in this case? I'm trying to implement a similar split structures but getting stuck in this problem. It would be a great help if you can help me solve this problem. Thank you very much.

P/s: Sorry for any mistakes. English is not my native language.

**Comment by avisekiit** [comment]
HI @tadax  : can I now use the repo, if you have corrected how the G network is updated.

**Comment by roxanneluo** [comment]
Hi @tadax, thank you a lot for sharing your code. Do you have any updates on this issue?

**Comment by RobinCSIRO** [comment]
I think it is easy to add the discriminator loss for training generator:
  def calc_g_loss(self, x, completion):
        loss1 = tf.nn.l2_loss(x - completion)
        loss2 = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=fake, labels=tf.ones_like(real)))
        loss = loss1+loss2
        return tf.reduce_mean(loss)

**Comment by fabrahman** [comment]
Hi @RobinCSIRO have you tried that? did you get reasonable results?
I guess he skiped one step in the algorithm. But then again for pretrainig step, we dont need the GAN loss to be incorporated.
Do we have to have two g_loss? one only the l2 norm an the other with l2+GANS_loss
